lua/lspconfig/volar.lua

```diff
@@ -0,0 +1,103 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local function get_typescript_server_path(root_dir)
+  local project_root = util.find_node_modules_ancestor(root_dir);
+  return project_root and (project_root .. '/node_modules/typescript/lib/tsserverlibrary.js') or '';
+end
+
+local server_name = 'volar'
+local bin_name = 'volar-server'
+
+-- https://github.com/johnsoncodehk/volar/blob/master/packages/shared/src/types.ts
+local volar_init_options = {
+  typescript = {
+    serverPath = get_typescript_server_path(vim.fn.getcwd()),
+  },
+  languageFeatures={
+    references = true,
+    definition = true,
+    typeDefinition = true,
+    callHierarchy = true,
+    hover = true,
+    rename = true,
+    renameFileRefactoring = true,
+    signatureHelp = true,
+    codeAction = true,
+    completion={
+      defaultTagNameCase='both',
+      defaultAttrNameCase='kebabCase',
+    },
+    schemaRequestService  = true ,
+    documentHighlight = true,
+    documentLink = true ,
+    codeLens = true,
+    semanticTokens = true,
```

Comment:
We don't support semanticTokens currently, or documentColor (if it is what I think it is

Reply:
Set both of those to false. Added links to relvant neovim PRs